src/main/java/teammates/ui/controller/InstructorCourseEditPageData.java

```diff
@@ -7,24 +8,129 @@
 import teammates.common.datatransfer.InstructorAttributes;
 import teammates.common.util.Const;
 import teammates.common.util.Url;
+import teammates.ui.template.CourseEditInstructorPanel;
+import teammates.ui.template.ElementTag;
 
 
 public class InstructorCourseEditPageData extends PageData {
-
-    public InstructorCourseEditPageData(AccountAttributes account) {
+    private int instructorToShowIndex;
+    private ElementTag deleteCourseButton;
+    private CourseAttributes course;
+    private List<CourseEditInstructorPanel> instructorPanelList;
+    private CourseEditInstructorPanel addInstructorPanel;
+    private InstructorAttributes currentInstructor;
+    private ElementTag addInstructorButton;
+    
+    public InstructorCourseEditPageData(AccountAttributes account, CourseAttributes course, 
+                                        List<InstructorAttributes> instructorList, 
+                                        InstructorAttributes currentInstructor, int instructorToShowIndex, 
+                                        List<String> sectionNames, List<String> feedbackNames) {
         super(account);
-        index = -1;
+        instructorPanelList = new ArrayList<CourseEditInstructorPanel>();
+        this.course = course;
+        this.instructorToShowIndex = instructorToShowIndex;
+        this.currentInstructor = currentInstructor;
+        
+        createButtons();
+        createInstructorPanelList(instructorList, sectionNames, feedbackNames);
```

Comment:
It should be best to connect `createInstructorPanelList()` with its object initialisation.
Try either moving the `instructorPanelList = new ArrayList<...` into the method, or do a `instructorPanelList = createInstructorPanelList(...` here like what was done for `addInstructorPanel` below.